src/npc_attack.cpp

```diff
@@ -486,23 +600,28 @@ npc_attack_rating npc_attack_throw::evaluate_tripoint(
     if( critter ) {
         att = source.attitude_to( *critter );
     }
-    const bool friendly_fire = att == Creature::Attitude::FRIENDLY &&
-                               !source.rules.has_flag( ally_rule::avoid_friendly_fire );
-    int attitude_mult = npc_attack_constants::attitude_multiplier.at( att );
-    if( friendly_fire ) {
-        // hitting a neutral creature isn't exactly desired, but it's a lot less than a friendly.
-        // if friendly fire is on, we don't care too much, though if an available hit doesn't damage them it would be better.
-        attitude_mult = npc_attack_constants::attitude_multiplier.at( Creature::Attitude::NEUTRAL );
+
+    if( att != Creature::Attitude::HOSTILE ) {
+        // No point in throwing stuff at neutral/friendly targets
+        return npc_attack_rating( cata::nullopt, location );
+    }
+
+    if( source.rules.has_flag( ally_rule::avoid_friendly_fire ) &&
+        !source.wont_hit_friend( location, thrown_item, false ) ) {
+        // Avoid friendy fire
+        return npc_attack_rating( cata::nullopt, location );
     }
 
     const float throw_mult = throw_cost( source, single_item ) * source.speed_rating() / 100.0f;
     const int damage = source.thrown_item_total_damage_raw( single_item );
     float dps = damage / throw_mult;
     const int distance_to_me = rl_dist( location, source.pos() );
-    const int distance_penalty = std::max( distance_to_me - 1,
-                                           ( source.closest_enemy_to_friendly_distance() - 1 ) * 2 );
+    const int distance_penalty = std::max( std::min( distance_to_me,
+                                           source.closest_enemy_to_friendly_distance() ) - 3, 0 );
```

Comment:
```suggestion
    const int closest_enemy_dist = source.closest_enemy_to_friendly_distance();
    int distance_penalty = std::abs( distance_to_me - 3) ;
    // Ideally, the 3 in the above should be adjusted for reach weapons, etc
    if ( closest_enemy_dist < distance_penalty + 2 ) { // not sure exactly how to adjust distance_penalty here
        distance_penalty = std::min( distance_penalty, std::abs( closest_enemy_dist - 1 ) * 2 );
    }
```
This will need considerable formatting! It penalizes throwing when something is close enough that melee should be preferred, or when throwing too close to a friendly.

Reply:
I'll add this change along with one of KorGgenT suggestions when I have access to my environment to test it, it will be in a few hours. The math still is a placeholder and will need more tweaking in a new PR. Hopefully from somene that can work it out as right now the modifiers for each kind of attack are inconsistent.